openbas-api/src/main/java/io/openbas/service/InjectService.java

```diff
@@ -1437,167 +1793,4 @@ private Double getValueAsDouble(Row row, String cellColumn) {
-
-    // Array aggregations
-    Expression<String[]> tagIdsExpression = createJoinArrayAggOnId(cb, injectRoot, "tags");
-    Expression<String[]> teamIdsExpression = createJoinArrayAggOnId(cb, injectRoot, "teams");
-    Expression<String[]> assetIdsExpression = createJoinArrayAggOnId(cb, injectRoot, "assets");
-    Expression<String[]> assetGroupIdsExpression =
-        createJoinArrayAggOnId(cb, injectRoot, "assetGroups");
-
-    // SELECT
-    cq.multiselect(
-            injectRoot.get("id").alias("inject_id"),
-            injectRoot.get("title").alias("inject_title"),
-            injectRoot.get("enabled").alias("inject_enabled"),
-            injectRoot.get("content").alias("inject_content"),
-            injectRoot.get("allTeams").alias("inject_all_teams"),
-            injectExerciseJoin.get("id").alias("inject_exercise"),
-            injectScenarioJoin.get("id").alias("inject_scenario"),
-            injectRoot.get("dependsDuration").alias("inject_depends_duration"),
-            injectorContractJoin.alias("inject_injector_contract"),
-            tagIdsExpression.alias("inject_tags"),
-            teamIdsExpression.alias("inject_teams"),
-            assetIdsExpression.alias("inject_assets"),
-            assetGroupIdsExpression.alias("inject_asset_groups"),
-            injectorJoin.get("type").alias("inject_type"),
-            injectDependency.alias("inject_depends_on"))
-        .distinct(true);
-
-    // GROUP BY
-    cq.groupBy(
-        Arrays.asList(
-            injectRoot.get("id"),
-            injectExerciseJoin.get("id"),
-            injectScenarioJoin.get("id"),
-            injectorContractJoin.get("id"),
-            injectorJoin.get("id"),
-            injectDependency.get("id")));
-  }
-
-  private List<InjectOutput> execInject(TypedQuery<Tuple> query) {
-    return query.getResultList().stream()
-        .map(
-            tuple ->
-                new InjectOutput(
-                    tuple.get("inject_id", String.class),
-                    tuple.get("inject_title", String.class),
-                    tuple.get("inject_enabled", Boolean.class),
-                    tuple.get("inject_content", ObjectNode.class),
-                    tuple.get("inject_all_teams", Boolean.class),
-                    tuple.get("inject_exercise", String.class),
-                    tuple.get("inject_scenario", String.class),
-                    tuple.get("inject_depends_duration", Long.class),
-                    tuple.get("inject_injector_contract", InjectorContract.class),
-                    tuple.get("inject_tags", String[].class),
-                    tuple.get("inject_teams", String[].class),
-                    tuple.get("inject_assets", String[].class),
-                    tuple.get("inject_asset_groups", String[].class),
-                    tuple.get("inject_type", String.class),
-                    tuple.get("inject_depends_on", InjectDependency.class)))
-        .toList();
-  }
 }
```

Comment:
**suggestion(non-blocking) :** This class is too long. We should think of splitting it ...